api/releases.go

```diff
@@ -0,0 +1,124 @@
+package api
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/ONSdigital/dp-search-api/query"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+// SearchReleasesHandlerFunc returns a http handler function handling release calendar search api requests.
+func SearchReleasesHandlerFunc(validator QueryParamValidator, builder ReleaseQueryBuilder, searcher ElasticSearcher, transformer ResponseTransformer) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		ctx := req.Context()
+		params := req.URL.Query()
+
+		q, err := url.QueryUnescape(params.Get("query"))
+		if err != nil {
+			log.Warn(ctx, err.Error(), log.Data{"param": "query", "value": params.Get("query")})
```

Comment:
(minor): this should be ```log.error``` and not ```warn``` ?

Reply:
Will merge now and will change both of these in next commit